### PR TITLE
add some details to field/group/scalar doc

### PR DESCRIPTION
**Pull request by bendyarm**
In order for a developer to know what their code is supposed to do, it helps to know the
exact spec of the language.  For field and scalar values, it helps to know the allowed values.
For the group, the definition of `+` in terms of coordinates would be too verbose here,
but at least stating that it is a Twisted Edwards curve with specific `a` and `d` will let
someone look up the details and do the calculations.  Also specified the generator point.
I do not have the environment to make sure the html builds right, so I am hoping a reviewer
can check that before merging this branch.

### CODE REVIEW

File: documentation/leo/03_language.md [hunk 1]
```diff
@@ -57,23 +57,26 @@ let b: u8 = 2; // implicit type -- not supported
 
 ### Field Elements
 
-Leo supports the `field` type for the base field elements of the elliptic curve.
-These are unsigned integers below the modulus of the base field.
+Leo supports the `field` type for elements of the base field of the elliptic curve.
+These are unsigned integers less than the modulus of the base field.  Showing the
```
Comment (acoglio) [inline documentation/leo/03_language.md:61]:
```suggestion
These are unsigned integers less than the modulus of the base field.  The following are the
```

**Review by acoglio — APPROVED** [review]

File: documentation/aleo/03_language.md [hunk 2]
```diff
@@ -103,7 +111,7 @@ program _foo.aleo;  // invalid
 
 ### Import
 
-An import is declared as `import {ProgramID};`.  
+An import is declared as `import {ProgramID};`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:114]:
```suggestion
An import is declared as `import {ProgramID};`.  
```

File: documentation/aleo/03_language.md [hunk 3]
```diff
@@ -115,7 +123,7 @@ program hello.aleo;
 
 ### Function
 
-A function is declared as `function {name}:`.  
+A function is declared as `function {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:126]:
```suggestion
A function is declared as `function {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 4]
```diff
@@ -129,18 +137,18 @@ function foo:
 
 #### Function Inputs
 
-A function input is declared as `input {register} as {type}.{visibility};`.  
+A function input is declared as `input {register} as {type}.{visibility};`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:140]:
```suggestion
A function input is declared as `input {register} as {type}.{visibility};`.  
```

File: documentation/aleo/03_language.md [hunk 5]
```diff
@@ -129,18 +137,18 @@ function foo:
...
 // The function `foo` takes a single input `r0` with type `field` and visibility `public`.
 function foo:
-    input r0 as field.public; 
+    input r0 as field.public;
 ```
 
 #### Function Outputs
 
-A function output is declared as `output {register} as {type}.{visibility};`.  
+A function output is declared as `output {register} as {type}.{visibility};`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:151]:
```suggestion
A function output is declared as `output {register} as {type}.{visibility};`.  
```

File: documentation/aleo/03_language.md [hunk 6]
```diff
@@ -150,13 +158,13 @@ Function outputs must be declared at the end of the function definition.
 
 #### Call a Function
 In the Aleo protocol, calling a function creates a transition that can consume and produce records on-chain.
-Use the `aleo run` CLI command to pass inputs to a function and execute the program.  
+Use the `aleo run` CLI command to pass inputs to a function and execute the program.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:161]:
```suggestion
Use the `aleo run` CLI command to pass inputs to a function and execute the program.  
```

File: documentation/aleo/03_language.md [hunk 7]
```diff
@@ -169,7 +177,7 @@ function call_external:
 
 ### Closure
 
-A closure is declared as `closure {name}:`.  
+A closure is declared as `closure {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:180]:
```suggestion
A closure is declared as `closure {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 8]
```diff
@@ -195,7 +203,7 @@ function call_internal:
 
 ### Struct
 
-A struct is a data type declared as `struct {name}:`.  
+A struct is a data type declared as `struct {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:206]:
```suggestion
A struct is a data type declared as `struct {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 9]
```diff
@@ -217,9 +225,9 @@ function new_array3:
 
 ### Record
 
-A [record](../concepts/02_records.md) type is declared as `record {name}:`.  
-Records contain component declarations `{name} as {type}.{visibility};`.  
-Record data structures must contain the `owner` declaration as shown below.  
+A [record](../concepts/02_records.md) type is declared as `record {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:228]:
```suggestion
A [record](../concepts/02_records.md) type is declared as `record {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 10]
```diff
@@ -217,9 +225,9 @@ function new_array3:
...
+Records contain component declarations `{name} as {type}.{visibility};`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:229]:
```suggestion
Records contain component declarations `{name} as {type}.{visibility};`.  
```

File: documentation/aleo/03_language.md [hunk 11]
```diff
@@ -217,9 +225,9 @@ function new_array3:
...
+Record data structures must contain the `owner` declaration as shown below.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:230]:
```suggestion
Record data structures must contain the `owner` declaration as shown below.  
```

File: documentation/aleo/03_language.md [hunk 12]
```diff
@@ -244,7 +252,7 @@ function new_token:
 
 ### Mapping
 
-A mapping is declared as `mapping {name}:`.  
+A mapping is declared as `mapping {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:255]:
```suggestion
A mapping is declared as `mapping {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 13]
```diff
@@ -309,13 +317,13 @@ A remove command that removes a key-value pair from a mapping, e.g. `remove acco
 
 ### Finalize
 
-A finalize is declared as `finalize {name}:`.  
+A finalize is declared as `finalize {name}:`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:320]:
```suggestion
A finalize is declared as `finalize {name}:`.  
```

File: documentation/aleo/03_language.md [hunk 14]
```diff
@@ -309,13 +317,13 @@ A remove command that removes a key-value pair from a mapping, e.g. `remove acco
...
 A finalize must immediately follow a [function](#function), and must have the same name;
 it is associated with the function and is executed on chain,
 after the zero-knowledge proof of the execution of the associated function is verified;
-a finalize *finalizes* a function on chain.  
-Upon success of the finalize function, the program logic is executed.  
-Upon failure of the finalize function, the program logic is reverted.  
+a finalize *finalizes* a function on chain.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:324]:
```suggestion
a finalize *finalizes* a function on chain.  
```

File: documentation/aleo/03_language.md [hunk 15]
```diff
@@ -309,13 +317,13 @@ A remove command that removes a key-value pair from a mapping, e.g. `remove acco
...
+Upon success of the finalize function, the program logic is executed.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:325]:
```suggestion
Upon success of the finalize function, the program logic is executed.  
```

File: documentation/aleo/03_language.md [hunk 16]
```diff
@@ -309,13 +317,13 @@ A remove command that removes a key-value pair from a mapping, e.g. `remove acco
...
+Upon failure of the finalize function, the program logic is reverted.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:326]:
```suggestion
Upon failure of the finalize function, the program logic is reverted.  
```

File: documentation/aleo/03_language.md [hunk 17]
```diff
@@ -361,7 +369,7 @@ finalize transfer_public_to_private:
 The following commands are supported in Aleo Instructions to provide additional program functionality.
 
 #### self.caller
-The `self.caller` command returns the address of the caller of the program.  
+The `self.caller` command returns the address of the caller of the program.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:372]:
```suggestion
The `self.caller` command returns the address of the caller of the program.  
```

File: documentation/aleo/03_language.md [hunk 18]
```diff
@@ -376,9 +384,9 @@ assert.eq block.height 100u64;
 
 #### rand.chacha
 
-The `rand.chacha` command returns a random number generated by the [ChaCha20 algorithm](http://cr.yp.to/chacha.html).  
-This command supports sampling a random `address`, `boolean`, `field`, `group`, `i8`, `i16`, `i32`, `i64`, `i128`, `u8`, `u16`, `u32`, `u64`, `u128`, and `scalar`.  
-Up to two additional seeds can be provided to the `rand.chacha` command.  
+The `rand.chacha` command returns a random number generated by the [ChaCha20 algorithm](http://cr.yp.to/chacha.html).
```
Comment (collinc97) [inline documentation/aleo/03_language.md:387]:
```suggestion
The `rand.chacha` command returns a random number generated by the [ChaCha20 algorithm](http://cr.yp.to/chacha.html).  
```

File: documentation/aleo/03_language.md [hunk 19]
```diff
@@ -376,9 +384,9 @@ assert.eq block.height 100u64;
...
+This command supports sampling a random `address`, `boolean`, `field`, `group`, `i8`, `i16`, `i32`, `i64`, `i128`, `u8`, `u16`, `u32`, `u64`, `u128`, and `scalar`.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:388]:
```suggestion
This command supports sampling a random `address`, `boolean`, `field`, `group`, `i8`, `i16`, `i32`, `i64`, `i128`, `u8`, `u16`, `u32`, `u64`, `u128`, and `scalar`.  
```

File: documentation/aleo/03_language.md [hunk 20]
```diff
@@ -376,9 +384,9 @@ assert.eq block.height 100u64;
...
+Up to two additional seeds can be provided to the `rand.chacha` command.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:389]:
```suggestion
Up to two additional seeds can be provided to the `rand.chacha` command.  
```

File: documentation/aleo/03_language.md [hunk 21]
```diff
@@ -418,7 +426,7 @@ Checkout the [Aleo Instructions opcodes](./04_opcodes.md) for a full list of sup
 
 #### Commit
 
-Aleo Instructions supports the following syntax for committing to standard types.  
+Aleo Instructions supports the following syntax for committing to standard types.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:429]:
```suggestion
Aleo Instructions supports the following syntax for committing to standard types.  
```

**Review by moAlobaidi — APPROVED** [review]
Technically on PTO rn, but ran this locally and HTML renders correctly. LGTM!

File: documentation/aleo/03_language.md [hunk 22]
```diff
@@ -433,8 +441,8 @@ commit.ped128 ...;
 Checkout the [Aleo Instructions opcodes](./04_opcodes.md) for a full list of supported commitment algorithms.
 
 #### position, branch.eq, branch.neq
-The `position` command, e.g. `position exit`, indicates a point to branch execution to.  
-The `branch.eq` command, e.g. `branch.eq r0 r1 to exit`, which branches execution to the position indicated by `exit` if `r0` and `r1` are equal.  
+The `position` command, e.g. `position exit`, indicates a point to branch execution to.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:444]:
```suggestion
The `position` command, e.g. `position exit`, indicates a point to branch execution to.  
```

File: documentation/aleo/03_language.md [hunk 23]
```diff
@@ -433,8 +441,8 @@ commit.ped128 ...;
...
+The `branch.eq` command, e.g. `branch.eq r0 r1 to exit`, which branches execution to the position indicated by `exit` if `r0` and `r1` are equal.
```
Comment (collinc97) [inline documentation/aleo/03_language.md:445]:
```suggestion
The `branch.eq` command, e.g. `branch.eq r0 r1 to exit`, which branches execution to the position indicated by `exit` if `r0` and `r1` are equal.  
```

File: documentation/leo/03_language.md [hunk 24]
```diff
@@ -46,7 +46,7 @@ Higher bit length integers generate more constraints in the circuit, which can s
 
 #### A Note on Leo Integers
 
-Leo will not default to an integer type. The definition of a integer **must** include an explicit type.  
+Leo will not default to an integer type. The definition of a integer **must** include an explicit type.
```
Comment (collinc97) [inline documentation/leo/03_language.md:49]:
```suggestion
Leo will not default to an integer type. The definition of a integer **must** include an explicit type.  
```

**Review by collinc97 — CHANGES_REQUESTED** [review]
My comments add back double spaces at the end of lines in markdown.
This is desired because it starts the next sentence on a new line in the documentation - improving readability.

I believe your editor removes double spaces at the end of a line @bendyarm. Please address the changes and disable the feature in your editor in the future.

**Comment by bendyarm** [comment]
@collinc97 Thanks for catching that.
I undid all the end-of-line space removals so the comparison now is just the changed sections.

**Review by collinc97 — APPROVED** [review]

**Pull request merged**